### PR TITLE
hide discard for nonauthors

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Discard change should be only possible for the author. BE already supports it, FE was too open.

Screenshot 2022-12-08 at 11 35 41

Also refactored components that were incorrectly placed in Overview instead of ChangeRequest itself

Screenshot 2022-12-08 at 11 50 09

### CODE REVIEW

File: frontend/src/component/changeRequest/ChangeRequest/Changes/Change/Discard.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    const { isChangeRequestConfigured } = useChangeRequestsEnabled(
+        changeRequest.project
+    );
+    const allowChangeRequestActions = isChangeRequestConfigured(
+        changeRequest.environment
+    );
+    const isPending = !['Cancelled', 'Applied'].includes(changeRequest.state);
+
+    const { user } = useAuthUser();
+    const isAuthor = user?.id === changeRequest.createdBy.id;
```
Comment (kwasniew) [inline frontend/src/component/changeRequest/ChangeRequest/Changes/Change/Discard.tsx:22]: this is the only new check

File: frontend/src/component/changeRequest/ChangeRequest/Changes/Change/Discard.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import React, { FC } from 'react';
```
Comment (kwasniew) [inline frontend/src/component/changeRequest/ChangeRequest/Changes/Change/Discard.tsx:1]: This file aggregates all existing code for Discard

File: frontend/src/component/changeRequest/ChangeRequest/ChangeRequest.tsx [hunk 3]
```diff
@@ -1,221 +1,25 @@
...
-        } catch (error: unknown) {
-            setToastApiError(formatUnknownError(error));
-        }
-    };
-
     return (
         <Box>
             {changeRequest.features?.map(feature => (
-                <ChangeRequestFeatureToggleChange
+                <FeatureToggleChanges
```
Comment (kwasniew) [inline frontend/src/component/changeRequest/ChangeRequest/ChangeRequest.tsx:22]: Cleaned up the component names and their placement in nested dirs

File: frontend/src/component/changeRequest/ChangeRequest/Changes/Change/Discard.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    },
+}));
+
+const Discard: FC<{ onDiscard: () => void }> = ({ onDiscard }) => (
+    <Box>
+        <StyledLink onClick={onDiscard}>Discard</StyledLink>
+    </Box>
+);
+
+export const DiscardContainer: FC<{
```
Comment (kwasniew) [inline frontend/src/component/changeRequest/ChangeRequest/Changes/Change/Discard.tsx:46]: This component aggregates all discard related functionality in one file

**Review by andreas-unleash — APPROVED** [review]
LGTM!

**Pull request merged**